src/app/page/Signup/SignupScene.jsx

```diff
@@ -46,59 +45,64 @@ function SignupScene(props) {
     };
   }
 
-  function handleSignupSuccess(user) {
-    setValues({
-      ...values,
+  async function handleSignupSuccess(user) {
+    let data = {
+      ...getPayload(),
       phoneNumber: user.phoneNumber,
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.CONNECT);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.CONNECT));
     return false;
   }
-  function handleConnectSuccess(user) {
-    setValues({
-      ...values,
+
+  async function handleConnectSuccess(user) {
+    let data = {
+      ...getPayload(),
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.PROFILE);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.PROFILE));
```

Comment:
Yeh so for these we could just call `User.update` directly.

two ways to handle
async/await:
```js
await User.update(data.uid, data);
setActiveTab(...)
```
promises:
```js
User.update(data.uid, data).then(() => setActiveTab(...));
```

I actually prefer the promise the version here as then you don't have to make this handler an async call.
Also not sure why these return `false` but we can leave handlers as void.

Reply:
Oh I see `updateUser` is supposed to be a general handler for all of these.. I'll comment over there

Reply:
I've always felt .then chaining was a little easier to read until you get into callback hell and its like .then().then().then().then().then()... but _then_ you have a different problem.

Reply:
Lol pretty much, as soon as you have multiple promises from different sources you should just use await*

*unless they can be done parallel lol

Reply:
I wasn't sure on the returning false either. I didn't know if it had something to do with the firebase callback so I left it. I can't find any documentation for `firebaseui` in the FirebaseAuthUI component.

Reply:
It's left as a wrapper but I switched everything to await.